synapse/http/servlet.py

```diff
@@ -143,45 +143,76 @@ def parse_string(
             is not one of those allowed values.
     """
     return parse_string_from_args(
-        request.args, name, default, required, allowed_values, param_type, encoding
+        request.args, name, default, required, allowed_values, encoding
     )
 
 
-def parse_string_from_args(
-    args,
-    name,
-    default=None,
-    required=False,
-    allowed_values=None,
-    param_type="string",
-    encoding="ascii",
-):
+def parse_string_value(value, allowed_values, name="", encoding="ascii") -> str:
```

Comment:
I think this is a helper function, so should probably be named `_parse_string_value`, and all the parameters are always given, so we don't need defaults.

```suggestion
def _parse_string_value(value: Union[str, bytes], allowed_values: Optional[Iterable[str]], name: str, encoding: Optional[str]) -> str:
```